pkg/protocol/chains.go

```diff
@@ -0,0 +1,180 @@
+package protocol
+
+import (
+	"cmp"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/iotaledger/hive.go/ds/reactive"
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/log"
+	"github.com/iotaledger/iota-core/pkg/model"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+// Chains is a subcomponent of the protocol that exposes the chains that are managed by the protocol and that implements
+// the chain switching logic.
+type Chains struct {
+	// Set contains all non-evicted chains that are managed by the protocol.
+	reactive.Set[*Chain]
+
+	// Main contains the main chain.
+	Main reactive.Variable[*Chain]
+
+	// HeaviestClaimedCandidate contains the candidate chain with the heaviest claimed weight.
+	HeaviestClaimedCandidate reactive.Variable[*Chain]
+
+	// HeaviestAttestedCandidate contains the candidate chain with the heaviest attested weight.
```

Comment:
```suggestion
	// HeaviestAttestedCandidate contains the candidate chain with the heaviest weight as checked by attestations. The chain has not been instantiated into an engine yet.
```

Reply:
Same as previous comments - lets's just merged it?